agent/agentssh/agentssh.go

```diff
@@ -428,6 +435,19 @@ func (s *Server) startPTYSession(session ptySession, magicTypeLabel string, cmd
 	return nil
 }
 
+func (s *Server) showServiceBanner(ctx context.Context, session io.Writer) error {
+	banner, err := s.GetServiceBanner(ctx)
```

Comment:
Consider checking `banner.Enabled` first to avoid a round-trip.

Since this will also slow down the start of every interactive SSH session by a server round-trip, should we try to push the change from the server and store it in the agent or poll it asynchronously? One option that comes to mind is piggy-backing on agent metadata.

Reply:
This sounds very reasonable to me!  It would love if we can avoid the round trip.  I will look into metadata.